beacon-chain/sync/validate_beacon_attestation.go

```diff
@@ -69,12 +72,36 @@ func (s *Service) validateCommitteeIndexBeaconAttestation(ctx context.Context, p
 		traceutil.AnnotateError(span, err)
 		return pubsub.ValidationIgnore
 	}
-	if !strings.HasPrefix(originalTopic, fmt.Sprintf(format, digest, att.Data.CommitteeIndex)) {
+	preState, err := s.chain.AttestationPreState(ctx, att)
+	if err != nil {
+		log.WithError(err).Error("Failed to retrieve pre state")
+		traceutil.AnnotateError(span, err)
+		return pubsub.ValidationIgnore
+	}
+	valCount, err := helpers.ActiveValidatorCount(preState, helpers.SlotToEpoch(att.Data.Slot))
+	if err != nil {
+		log.WithError(err).Error("Could not retrieve active validator count")
+		traceutil.AnnotateError(span, err)
+		return pubsub.ValidationIgnore
+	}
+	subnet := helpers.ComputeSubnetForAttestation(valCount, att)
+
+	if !strings.HasPrefix(originalTopic, fmt.Sprintf(format, digest, subnet)) {
+		return pubsub.ValidationReject
+	}
+
+	// Attestation aggregation bits must exist.
```

Comment:
Please move this farther up the validation chain. We should order validation from least expensive checks first to most expensive checks last.

Reply:
I will fix this.